chore: update RELEASE.md with instructions (#6519)
prometheus-operator/prometheus-operator: RELEASE.md

|---------|----------------------------------|-----------------------------------------|
| v0.76   | 2024-08-07                       | **searching for volunteer**             |
| v0.75   | 2024-06-26                       | **searching for volunteer**             |
| v0.74   | 2024-05-15                       | Simon Pasquier (GitHub: @simonpasquier) |
| v0.73   | 2024-04-03                       | Jayapriya Pai (Github: @slashpai)       |
| v0.72   | 2024-02-21                       | Arthur Sens (Github: @ArthurSens)       |
| v0.71   | 2024-01-10                       | Simon Pasquier (GitHub: @simonpasquier) |
| v0.70   | 2023-11-29                       | Pawel Krupa (GitHub: @paulfantom)       |
| v0.69   | 2023-10-18                       | Simon Pasquier (GitHub: @simonpasquier) |
| v0.68   | 2023-09-06                       | Arthur Sens (Github: @ArthurSens)       |
| v0.67   | 2023-07-26                       | Simon Pasquier (GitHub: @simonpasquier) |
| v0.66   | 2023-06-14                       | Arthur Sens (Github: @ArthurSens)       |
| v0.65   | 2023-05-03                       | Philip Gough (GitHub: @PhilipGough)     |

> 

Author: @simonpasquier removed old entries as per suggestion

Following [Prometheus](https://github.com/prometheus/prometheus/blob/main/RELEASE.md) and [Thanos](https://github.com/thanos-io/thanos/blob/main/docs/release-process.md), this project aims for a predictable release schedule.

The release cycle for cutting releases is every 6 weeks

| Release | Date of release (year-month-day) | Release shepherd                        |
|---------|--------------------------------------------|-----------------------------------------|
| v0.75   | 2024-06-26                                 | **searching for volunteer**             |

> | v0.74   | 2024-05-15                                 | **searching for volunteer**             |

Reviewer: can you add me there :)

| v0.73   | 2024-04-03                                 | Jayapriya Pai (Github: @slashpai)       |
| v0.72   | 2024-02-21                                 | Arthur Sens (Github: @ArthurSens)       |
| v0.71   | 2024-01-10                                 | Simon Pasquier (GitHub: @simonpasquier) |
| v0.70   | 2023-11-29                                 | Pawel Krupa (GitHub: @paulfantom)       |
| v0.69   | 2023-10-18                                 | Simon Pasquier (GitHub: @simonpasquier) |
| v0.68   | 2023-09-06                                 | Arthur Sens (Github: @ArthurSens)       |
| v0.67   | 2023-07-26                                 | Simon Pasquier (GitHub: @simonpasquier) |
| v0.66   | 2023-06-14                                 | Arthur Sens (Github: @ArthurSens)       |

> | v0.65   | 2023-05-03                                 | Philip Gough (GitHub: @PhilipGough)     |

Reviewer: I would also remove all entries from there. From my POV, the idea of keeping the history is to have a rough idea of who did the release in the past and have a healthy rotation.

Author: done

Now that all version information has been updated, an entry for the new version can be added to the `CHANGELOG.md` file.

> Note that CHANGELOG.md should only document changes relevant to users of prometheus-operator, including external API changes, performance improvements, and new features. Do not document changes of internal interfaces, code refactorings and clean-ups, changes to the build process, etc. People interested in these are asked to refer to the git history.

Reviewer: ```suggestion
Note that CHANGELOG.md should only document changes relevant to users of prometheus-operator, including external API changes, performance improvements, and new features. Do not document changes of internal interfaces, code refactoring and clean-ups, changes to the build process, etc.
```

Create a PR for the changes to be reviewed.

> You can use github UI to get differences between commits in latest release branch and previous released version.

Reviewer: ```suggestion
You can use the GitHub UI to see the difference between the release branch and the latest stable release.
```

> eg: https://github.com/prometheus-operator/prometheus-operator/compare/v0.72.0...release-0.73

Reviewer: ```suggestion
For example: https://github.com/prometheus-operator/prometheus-operator/compare/v0.72.0...release-0.73

Unless exception, the latest tag shouldn't contain commits that don't exist in the release branch.
``

## Publish the new release

For new minor and major releases, create the `release-<major>.<minor>` branch starting at the PR merge commit.
Push the branch to the remote repository with

> **Note:** The remote name `origin` is assumed to be pointed to `prometheus-operator/prometheus-operator`. If you have a different remote name, use that instead of `origin`. Verify this using `git remote -v`

Reviewer: ```suggestion
**Note:** The remote name `origin` is assumed to be pointed to `github.com/prometheus-operator/prometheus-operator`. If you have a different remote name, use that instead of `origin`. Verify this using `git remote -v`
```

Signed tag with a GPG key is appreciated, but in case you can't add a GPG key to your Github account using the following [procedure](https://docs.github.com/articles/generating-a-gpg-key), you can replace the `-s` flag by `-a` flag of the `git tag` command to only annotate the tag without signing.

> Once a tag is created, the release process through Github action will be triggered for this tag. Our CI pipeline will automatically push the container images to [quay.io](https://quay.io/organization/prometheus-operator) and [ghcr.io](https://github.com/prometheus-operator/prometheus-operator/pkgs/container/prometheus-operator). Wait till [publish](https://github.com/prometheus-operator/prometheus-operator/actions/workflows/publish.yaml) and [release](https://github.com/prometheus-operator/prometheus-operator/actions/workflows/release.yaml) jobs are completed so that assets will be attached to release correctly. We have observed in past that if we create a draft release and publish it later assets are not attached correctly hence its advised to wait till release job is completed to create the release.

Reviewer: I feel that the last sentence should be moved to the next paragraph where we instruct how to create/publish the release.
```suggestion
Once a tag is created, the `publish` Github action will push the container images to [quay.io](https://quay.io/organization/prometheus-operator) and [ghcr.io](https://github.com/prometheus-operator/prometheus-operator/pkgs/container/prometheus-operator). Wait until the [publish](https://github.com/prometheus-operator/prometheus-operator/actions/workflows/publish.yaml) workflow is complete before going to the next step.
```

Signed tag with a GPG key is appreciated, but in case you can't add a GPG key to your Github account using the following [procedure](https://docs.github.com/articles/generating-a-gpg-key), you can replace the `-s` flag by `-a` flag of the `git tag` command to only annotate the tag without signing.

> Once a tag is created, the release process through github actions will be triggered for this tag. Our CI pipeline will automatically push the container images to [quay.io](https://quay.io/organization/prometheus-operator) and [ghcr.io](https://github.com/prometheus-operator/prometheus-operator/pkgs/container/prometheus-operator). Wait till [publish](https://github.com/prometheus-operator/prometheus-operator/actions/workflows/publish.yaml) and [release](https://github.com/prometheus-operator/prometheus-operator/actions/workflows/release.yaml) jobs are completed so that assets will be attached to release correctly.

Reviewer: I'm not sure about the second part: IIRC the release job will not kick in before we create the new release but yes it's best to wait for the publish to complete.

```suggestion
Once a tag is created, the release a GitHub action will push the container images to [quay.io](https://quay.io/organization/prometheus-operator) and [ghcr.io](https://github.com/prometheus-operator/prometheus-operator/pkgs/container/prometheus-operator). Wait till [publish](https://github.com/prometheus-operator/prometheus-operator/actions/workflows/publish.yaml) and [release](https://github.com/prometheus-operator/prometheus-operator/actions/workflows/release.yaml) jobs are completed so that assets will be attached to release correctly.
```

> Go to https://github.com/prometheus-operator/prometheus-operator/releases/new, associate the new release with the before pushed tag, paste in changes made to `CHANGELOG.md` and click "Publish release".

Reviewer: Do you want to add that we shouldn't use Draft mode? Or do you want to try fixing the action?
